tests/lmeval/test_lmeval.py

```diff
@@ -152,46 +193,124 @@ def _run_lm_eval(self):
             batch_size=self.lmeval.batch_size,
         )
 
+        # Always use recovery testing
+        self._validate_recovery(results)
+
+        # If absolute metrics provided, show warnings (not failures)
+        if self.lmeval.metrics:
+            self._check_absolute_warnings(results)
+
+    def _validate_recovery(self, compressed_results):
+        """Validate using recovery testing - compare against base model."""
+        base_metrics = self.base_results["results"][self.lmeval.task]
+        compressed_metrics = compressed_results["results"][self.lmeval.task]
+        higher_is_better_map = compressed_results["higher_is_better"][self.lmeval.task]
+
+        # Get recovery threshold(s)
+        recovery_threshold = self.lmeval.recovery_threshold
+        is_dict = isinstance(recovery_threshold, dict)
+
+        logger.info("=" * 80)
+        logger.info("RECOVERY TESTING COMPARISON")
+        logger.info("=" * 80)
+
+        failures = []
+        for metric_key, base_val in base_metrics.items():
+            # Skip stderr and other metadata
+            if "stderr" in metric_key or metric_key.startswith("alias"):
+                continue
```

Comment:
Shouldn't we be matching metrics to results, rather than results to metrics? That way we don't have to hard code against stuff like this.

```
for metric_key in compressed_metrics:
     ...

for metric_key in recovery_threshold:
     ....
```